src/formulas/composer_tokenizer.ts

```diff
@@ -30,19 +30,20 @@ interface FunctionContext {
  *
  * The information added are:
  * - start, end and length of each token
- * - range detection (replaces the tokens that composes the range with 1 token)
- * - parenthesis matching (only for parenthesis tokens)
- * - parent function (only for tokens surrounded by a function)
- * - arg position (only for tokens surrounded by a function)
+ * - parenthesesCode (a code indicating the position of the token in the parentheses tree)
+ * - functionContext (only for tokens surrounded by a function)
+ * - color (the base color of the token, without fading modification)
+ * - isFormulaCursorContextToken (indicates if the token is on the same formula as the cursor)
  */
 
 export interface EnrichedToken extends Token {
   start: number;
   end: number;
   length: number;
-  parenIndex?: number;
+  parenthesesCode?: string;
   functionContext?: FunctionContext;
   color?: Color;
+  isFormulaCursorContextToken?: boolean;
```

Comment:
I'd put `isHighlighted` or something. `isFormulaCursorContextToken` is a bit unclear, and the outside world doesn't care why the token is highlighted or not

Reply:
Or maybe the compsoer store could directly change the color

Reply:
what do you think of isBlurred ?
highlighted will be redundant with the ranges which are already highlighted on the colors